### PR TITLE
RavenDB-9686: seperate Database Cancel Operation into : query,

### PR DESCRIPTION
**Pull request by aviviadi**
index and deleteDocs parts

### CODE REVIEW

File: src/Raven.Server/Config/Categories/DatabaseConfiguration.cs [hunk 1]
```diff
@@ -6,11 +6,23 @@ namespace Raven.Server.Config.Categories
 {
     public class DatabaseConfiguration : ConfigurationCategory
     {
-        [Description("The time to wait before canceling a database operation such as load (many) or query")]
+        [Description("The time to wait before canceling query operation")]
         [DefaultValue(5)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/DatabaseConfiguration.cs]: Let's change the timing all to seconds, minutes is to big a granularity

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Config/Categories/DatabaseConfiguration.cs [hunk 2]
```diff
@@ -6,11 +6,23 @@ namespace Raven.Server.Config.Categories
...
+        [ConfigurationEntry("Databases.QueryOperationTimeoutInSec", ConfigurationEntryScope.ServerWideOrPerDatabase)]
+        public TimeSetting QueryOperationTimeout { get; set; }
+
+        [Description("The time in seconds to wait before canceling terms indexing operation")]
+        [DefaultValue(300)]
+        [TimeUnit(TimeUnit.Seconds)]
+        [ConfigurationEntry("Databases.IndexTermsOperationTimeoutInSec", ConfigurationEntryScope.ServerWideOrPerDatabase)]
+        public TimeSetting IndexTermsOperationTimeout { get; set; }
+
+        [Description("The time in seconds to wait before canceling indexing terms operation")]
```
Comment (arekpalinski) [inline src/Raven.Server/Config/Categories/DatabaseConfiguration.cs:21]: Wrong description.

Comment (arekpalinski) [reply src/Raven.Server/Config/Categories/DatabaseConfiguration.cs:21]: The usage of this setting is only when deleting docs from the studio - not sure if it really deserves the configuration option.

Did you intent to introduce it in for delete / patch by query operations?

Comment (aviviadi) [reply src/Raven.Server/Config/Categories/DatabaseConfiguration.cs:21]: Please see my question to you at http://issues.hibernatingrhinos.com/issue/RavenDB-9701
We need to modify this, I know.

Comment (arekpalinski) [reply src/Raven.Server/Config/Categories/DatabaseConfiguration.cs:21]: Answered there

Comment (aviviadi) [reply src/Raven.Server/Config/Categories/DatabaseConfiguration.cs:21]: fixed in https://github.com/ravendb/docs/pull/407

**Pull request merged**